e2e/__tests__/retainAllFiles.test.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import runJest from '../runJest';
+
+describe('Retain All Files Integration', () => {
+  test('valid test within node_modules', () => {
+    const config = {
+      haste: {
+        retainAllFiles: true,
+      },
+    };
+
+    const {exitCode} = runJest('retain-all-files', [
```

Comment:
should we get stdout as well and snapshot it, so we know the correct test was run?